### DISCUSSION TITLE
Micronaut Framework 4.4.0 Plan

### DISCUSSION THREAD

**Discussion by sdelamo** [micronaut-projects/micronaut-core#10527] [Planning]
On February 6, 2024, we released [4.3.0](https://micronaut.io/2024/02/06/micronaut-framework-4-3-0-released/).
I have scheduled Micronaut Framework 4.4.0 for March 19th, 2024

# 4.4.0 Plan
### Youtube Channel
Release ten videos of less than 15m for the YouTube channel covering features of the framework. We have to improve our presence on YouTube. Many people only learn with video nowadays.
### Guides
Release 4 Guides comparing Micronaut Framework to Spring Boot. It is vital to show how easy it is to move from Spring Boot to Micronaut Framework.
## New Modules
I think we may be able to release one new module
[SourceGen](https://github.com/micronaut-projects/micronaut-sourcegen/issues). 

> Micronaut SourceGen provides a fork for JavaPoet with enhanced features and abstraction for performing source code generation across languages.
## New Functionality
[Integrate with HTMX](https://github.com/micronaut-projects/micronaut-views/issues/660)

## Backlog
The following features are in the backlog. I don’t think we have time to complete this cycle. 
Data
- [Support Multitenancy via discriminator column](https://github.com/micronaut-projects/micronaut-data/issues/2577)
- [Integration with synk](https://github.com/micronaut-projects/micronaut-core/issues/10100)
AWS
- [Better integration with DynamoDB](https://github.com/micronaut-projects/micronaut-aws/pull/1664). 
- [Changes to Micronaut Launch](https://github.com/micronaut-projects/micronaut-starter/issues/2191)
Views
- [Scaffolding Feature](https://github.com/micronaut-projects/micronaut-views/issues/606)
Core
- [Block detection for methods annotated with `@Blocking`](https://github.com/micronaut-projects/micronaut-core/issues/10020)
Security
- [Support Paseto tokens](https://github.com/micronaut-projects/micronaut-security/pull/813)
GraphQL
- [Apollo GraphQL Federation support](https://github.com/micronaut-projects/micronaut-graphql/issues/398)
Micronaut CRaC 
- [Consider exposing an endpoint that creates a CRaC checkpoint programmatically when invoked](https://github.com/micronaut-projects/micronaut-crac/issues/116)
New Modules/Integrations
- [Load Shedding](https://github.com/micronaut-projects/micronaut-core/issues/9293)
- [Micronaut Pay](https://github.com/micronaut-projects/micronaut-core/issues/10134)
- [OpenSearch Module](https://github.com/micronaut-projects/micronaut-elasticsearch/issues/443)
- [rSocket support](https://github.com/micronaut-projects/micronaut-core/issues/2002)
- [Integration with Langchain4j](https://github.com/micronaut-projects/micronaut-core/issues/10009)
- [Better Integration with GraalVM Truffle](https://github.com/micronaut-projects/micronaut-core/issues/10012)

any feedback is welcome!

**Comment by nedelva** [comment]
Speaking of Advocacy and Guides: are there any write-ups showcasing transition from Spring Boot to Micronaut? Any success stories from prominent clients? What are your compelling arguments for doing such a move? 

I have a hard time trying to convince some people from my organization to adopt Micronaut..

**Comment by sdelamo** [comment]
We need more time for 4.4.0. Target date is now April 4th, 2024.

**Comment by sdelamo** [comment]
# Micronaut Framework 4.4.0 Plan Summary 

[Micronaut Framework 4.4.0 announcement's blog post](https://micronaut.io/2024/04/17/micronaut-framework-4-4-0-released/)

## Advocacy
❌Youtube Channel - Ten Videos. We released only one. 
❌ Guides comparing Micronaut Framework to Spring Boot.

## Development
❌ [Micronaut SourceGen](https://github.com/micronaut-projects/micronaut-sourcegen) - Hopefully, we can release this in 4.5.0
✅ [HTMX Support](https://micronaut-projects.github.io/micronaut-views/5.2.1/guide/#htmx) 

## Unplanned
✅ [Micronaut OpenSearch](https://github.com/micronaut-projects/micronaut-opensearch) - We did not plan to do OpenSerach for 4.4.0, but we had an initial community contribution to kick off the development, and we released Micronaut OpenSearch as a new module.
